functions/src/export-csv.ts

```diff
@@ -145,130 +170,149 @@ function writeRow(
+  row.push(quote(submission.created?.emailAddress));
   csvStream.write(row);
 }
 
-/**
- * Returns the WKT string converted from the given geometry object
- *
- * @param geometryObject - the GeoJSON geometry object extracted from the LOI. This should have format:
- *   {
- *      coordinates: any[],
- *      type: string
- *   }
- * @returns The WKT string version of the object
- * https://www.vertica.com/docs/9.3.x/HTML/Content/Authoring/AnalyzingData/Geospatial/Spatial_Definitions/WellknownTextWKT.htm
- *
- * @beta
- */
-function toWkt(geometryObject: any): string {
-  return geojsonToWKT(Datastore.fromFirestoreMap(geometryObject));
+function quote(value: any): string {
+  if (value === null || value === undefined) {
+    return '';
+  }
+  if (typeof value === 'number') {
+    return value.toString();
+  }
+  const escaped = value.toString().replaceAll('"', '""');
+  return '"' + escaped + '"';
+}
+
+function toWkt(geometry: Pb.IGeometry): string {
+  return geojsonToWKT(toGeoJsonGeometry(geometry));
 }
 
 /**
- * Returns the string representation of a specific task element result.
+ * Returns the string or number representation of a specific task element result.
  */
-function getValue(taskId: string, task: Task, data: any) {
-  const result = data[taskId] || '';
-  if (
-    task.type === 'multiple_choice' &&
-    Array.isArray(result) &&
-    task.options
-  ) {
-    return result.map(id => getMultipleChoiceValues(id, task)).join(', ');
-  } else if (task.type === 'capture_location') {
-    if (!result) {
-      return '';
-    }
-    return toWkt(result.geometry || result);
+function getValue(
+  taskId: string,
+  task: Task,
+  data: Pb.ITaskData[]
+): string | number | null {
+  const result = data.find(d => d.taskId === taskId);
+  if (!result) {
+    return null;
+  }
+  if (result.textResponse) {
+    return result.textResponse.text || null;
+  } else if (result.numberResponse) {
+    return getNumberValue(result.numberResponse);
+  } else if (result.dateTimeResponse) {
+    return getDateTimeValue(result.dateTimeResponse);
+  } else if (result.multipleChoiceResponses) {
+    return getMultipleChoiceValues(task, result.multipleChoiceResponses);
+  } else if (result.captureLocationResult) {
+    // TODO(#1916): Include altitude and accuracy in separate columns.
+    return toWkt(
+      new Pb.Geometry({
+        point: new Pb.Point({
+          coordinates: result.captureLocationResult.coordinates,
+        }),
+      })
+    );
+  } else if (result.drawGeometryResult?.geometry) {
+    // TODO(#1248): Test when implementing other plot annotations feature.
+    return toWkt(result.drawGeometryResult.geometry);
+  } else if (result.takePhotoResult) {
+    return getPhotoUrlValue(result.takePhotoResult);
   } else {
-    return result;
+    return null;
   }
 }
 
+function getNumberValue(response: Pb.TaskData.INumberResponse): number | null {
+  const number = response.number;
+  if (number === undefined || number === null) {
```

Comment:
Using non-strict equality, i.e., `number != null`, here is the one time that it's actually useful!

Or you can simplify this whole expression to:
```
return response.number ?? null 
```

Reply:
Nice!